src/resetprg.c

```diff
@@ -100,20 +100,20 @@ void _fini(void) {
 	// empty
 }
 
-static void emptySection(uint32_t* start, uint32_t* end) {
-	uint32_t* dst = start;
+static void emptySection(uintptr_t start, uintptr_t end) {
+	uintptr_t dst = start;
 	while (dst < end) {
-		*dst = 0;
+		dst = 0;
 		++dst;
 	}
 }
 
-static void relocateSDRAMSection(uint32_t* start, uint32_t* end) {
-	uint32_t* src = (uint32_t*)((uint32_t)&__heap_start + ((uint32_t)start - PLACEMENT_SDRAM_START));
-	uint32_t* dst = start;
+static void relocateSDRAMSection(uintptr_t start, uintptr_t end) {
```

Comment:
These should still be pointers - they're the addresses of the start and end of the section. I think they'd switch to uint* if anything